nexus-and-alo/nexus-strawman.md

```diff
@@ -0,0 +1,361 @@
+**Contents:**
+
+- [Terminology](#terminology)
+- [User flows](#user-flows)
+  - [Hosting a Nexus Service](#hosting-a-nexus-service)
+  - [Consuming a Nexus Service from Temporal](#consuming-a-nexus-service-from-temporal)
+- [Concepts](#concepts)
+  - [Nexus Instantiation](#nexus-instantiation)
+  - [Nexus CRUD](#nexus-crud)
+  - [Nexus Task hosting](#nexus-task-hosting)
+  - [Nexus Task Worker](#nexus-task-worker)
+  - [Calling Nexus from Temporal](#calling-nexus-from-temporal)
+
+## Terminology
+
+**Nexus Service Instance:** Formally registered Nexus Service with a Nexus backend (eg: Temporal).  Nexus Instances must always have a unique identifier.
+
+**Nexus Task:** Unit of work which is run as a result of a request reaching the Nexus endpoint. Nexus Tasks are developed and operated similar to WorkflowTasks and ActivityTasks (eg: run and hosted on Nexus Workers). Within a Nexus Task Handler, developers can transform incoming payloads, make routing decisions and more.
+
+**Nexus Task Handler:** Source code which is designed to handle incoming Nexus Tasks. Nexus Task Handler is similar to a Workflow or Activity definition and is hosted by a Nexus Worker. 
+
+**Nexus Worker:** A special purpose Worker that pulls Nexus Tasks for its registered Nexus Instances from the Server and executes them using associated Nexus Task Handlers. 
+
+## User flows
+
+### Hosting a Nexus Service
+
+1. User instantiates Nexus Service Instance using provided `nexus` CLI commands (also possible via SDK or API) and receives an Instance ID. Nexus is now able to receive requests to its frontend endpoint (though they don't get processed until step 4).
```

Comment:
This is inescapable because we cannot guarantee global uniqueness in many realistic contexts. The Cloud is a great example of this. Is your Nexus name unique to the Namespace? Account? Cluster?

Reply:
> Is your Nexus name unique to the Namespace? Account? Cluster?

Endpoint. When I call service "foo" on "bar.temporal.cloud", that's all I need. An ID has nothing to do with it.

Reply:
I see, for me endpoint is one valid form of id. In other words we are more aligned than not.